docs: Cluster peering with Transparent Proxy updates (#14369)
hashicorp/consul: website/content/docs/connect/cluster-peering/k8s.mdx

- At least two Kubernetes clusters
- The installation must be running on Consul on Kubernetes version 0.47.1 or later

> ### Pre-setup 
> 
> 1. After provisioning a Kubernetes cluster and setting up your kubeconfig file to manage access to multiple Kubernetes clusters, export the Kubernetes context names for future with `kubectl`. 
> 

Reviewer: ```suggestion
### Pre-setup 

1. After provisioning a Kubernetes cluster and setting up your kubeconfig file to manage access to multiple Kubernetes clusters, export the Kubernetes context names for future with `kubectl`. 

```
First, is this a prerequisite or an additional step to the overall workflow? Calling it a pre-step (IMO) is little confusing. To keep it simple, this is either a prerequisite or the first step in the procedure.

Also, "After provisioning a Kubernetes cluster" tracks with the second bullet in the prerequisites section ("At least two Kubernetes clusters"), but "[After]. . . setting up your kubeconfig file to manage access to multiple Kubernetes clusters" is new. I think we should make this step it's own action either as part of the prerequisites or as the first (second?) step in the overall procedure.

Does that make sense?

Author: It is not an explicit pre-req but makes the overall process of issuing commands a lot easier if you are working with multiple clusters.

>   You can discover the context names using either `kubectl config current-context` for the current cluster you are on, or by issuing `kubectl config get-contexts` to get all configured contexts for your kubeconfig file.  

Reviewer: ```suggestion
You can use the following methods to get the context names for your clusters:
* Issue the `kubectl config current-context` command to get the context for the cluster you are currently in.
* Issue the `kubectl config get-contexts` command to get all configured contexts in your kubeconfig file.  
```
Is that any cleaner?

  ```shell-session
  $ export CLUSTER1_CONTEXT=<CONTEXT for first Kubernetes cluster>
  $ export CLUSTER2_CONTEXT=<CONTEXT for second Kubernetes cluster>
  ```

### Helm chart configuration

1. To establish cluster peering through Kubernetes, create a values.yaml file with the following Helm values. 
   

>   With these values,
>   the servers in each cluster will be exposed over a Kubernetes Load balancer service. This service can be customized
>   using [`server.exposeService`](/docs/k8s/helm#v-server-exposeservice). 

Reviewer: ```suggestion
  These values expose the servers in each cluster will over a Kubernetes Load balancer service. You can configure the [`server.exposeService`](/docs/k8s/helm#v-server-exposeservice) value to customize the service.
```
The more I think about, the more I think that we should position the content under "Pre-setup" and "Helm chart configuration" as the first step in the procedure with a heading like "Prepare your Kubernetes environment". We would just need to indicate that the practitioner only needs to perform this part of the procedure once (right?).

Author: I've added preparation for install as I believe that is the better way to describe the retrieval of kube contexts and creation of the values file under one heading.

>   When generating a peering token from one of the clusters, the address(es) of the load balancer will be used in the peering token, so the peering stream will go through the load balancer in front of the servers. For customizing the addresses used in the peering token, see [`global.peering.tokenGeneration`](/docs/k8s/helm#v-global-peering-tokengeneration).

Reviewer: ```suggestion
  When generating a peering token from one of the clusters, Consul uses the address(es) of the load balancer in the peering token so that the peering stream goes through the load balancer in front of the servers. For customizing the addresses used in the peering token, refer to [`global.peering.tokenGeneration`](/docs/k8s/helm#v-global-peering-tokengeneration).
```
Switched to present tense and active voice per the style guide. Also, do we mean "While generating a peering token . . ." in the first sentence? IF the operation can happen more than once, then we should say "While . . .", but if it is a one-time operation, then "When . . ." is OK.

- At least two Kubernetes clusters
- The installation must be running on Consul on Kubernetes version 0.47.1 or later

###  Preparation for install

> 1. After provisioning a Kubernetes cluster and setting up your kubeconfig file to manage access to multiple Kubernetes clusters, export the Kubernetes context names for future use with `kubectl`. 

Reviewer: ```suggestion
1. After provisioning a Kubernetes cluster and setting up your kubeconfig file to manage access to multiple Kubernetes clusters, export the Kubernetes context names for future use with `kubectl`. 
```
Should "Your kubeconfig file should be set up to manage access to multiple Kubernetes clusters" be one of the bullets under the main Prerequisites section? It is a new step that we are assuming the user already did by this point.

Author: I think its ok to leave this to the user but I'll add a link to the k8s docs to make this more explicit and reduce confusion.

1. Apply the intentions to the second cluster.

    ```shell-session
    $ kubectl --context $CLUSTER2_CONTEXT apply --filename intention.yml
    ```

> 1. For the services in `cluster-01` that you want to access the "backend-service," add the following annotations to the service file. In order dial the upstream service from an application, ensure that the requests are sent to the correct DNS name as specified in [Service Virtual IP Lookups](https://www.consul.io/docs/discovery/dns#service-virtual-ip-lookups). 

Reviewer: ```suggestion
1. For the services in `cluster-01` that you want to access the "backend-service," add the following annotations to the service file. To dial the upstream service from an application, ensure that the requests are sent to the correct DNS name as specified in [Service Virtual IP Lookups](https://www.consul.io/docs/discovery/dns#service-virtual-ip-lookups). 
```
